mlflow/lightgbm.py

```diff
@@ -483,6 +483,18 @@ def __init__(original, self, *args, **kwargs):
         original(self, *args, **kwargs)
 
     def train(_log_models, original, *args, **kwargs):
+        from mlflow.sklearn import _SklearnTrainingSession
+
+        # If this function is called within a session initiated by a scikit-learn estimator,
+        # logging the parameters passed in this function fails when the the scikit-learn
+        # estimator has overlapped parameters.
+        # See https://github.com/mlflow/mlflow/issues/7402 for more details.
+        if any(
+            x.estimator.__class__.__module__.startswith("sklearn.")
+            for x in _SklearnTrainingSession._session_stack
+        ):
```

Comment:
This condition, although it works for current code, it has a logical issue:

suppose in future, we allow sklearn pipeline to have nested autologging, then for sklearn pipeline, the training session `allow_children=True`, in this case, the stack is:

[ session(estimator=sklearn_pipeline, allow_children=True), session(estimator=lgbm, allow_children=False), session(estimator=lgbm, allow_children=False)]

in this case, the train function should enable lgbm autologging after checking the stack status.